Migrate abstraction alert tests

Cruikshanks opened the pull request:
This change migrates the legacy water abstraction alert acceptance tests. These can be found in [cypress/integration/internal/waa](https://github.com/DEFRA/water-abstraction-ui/tree/main/cypress/integration/internal/waa).

Again as previously found, the existing tests contain a mix of testing the journey and _some_ validations. To make it easier to asses their value going forward the 2 have been split.

In this particular case the legacy code appeared to contain 2 distinct tests; `gauging-stations-and-water-abstraction-alerts-mbod.spec` and `gauging-stations-and-water-abstraction-alerts.spec`. However, after review, they are almost identical. The only difference is in the choice of unit of measure for the threshold, **mBOD** or **Ml/d**. Whichever you select makes no difference to the subsequent journey. So, as part of the migration we have dropped the test for **Ml/d** as it is adding no value.

timstone commented on features/step_definitions/digitise_steps.rb:
```diff
@@ -0,0 +1,131 @@
+  @front_app.licences_page.nav_bar.ar_link.click
+  expect(@front_app.digitise_page.heading).to have_text("Review licence data")
+
+  # Failsafe to prevent test running in production
```
see above comment

andrewhick replied:
Done as above

timstone commented on features/step_definitions/digitise_steps.rb:
```diff
@@ -0,0 +1,131 @@
+
+  # Count the number of In Progress, Approved, Licence review after first edit
+  table_content = @front_app.digitise_review_page.licence_table.text
+  @no_of_in_progress = table_content.scan(/(?=#{"In progress"})/).count
```
You might want to have count methods in the page object rather that in the tests

andrewhick replied:
Done - looks neater, thanks.

timstone commented on features/step_definitions/digitise_steps.rb:
```diff
@@ -0,0 +1,131 @@
+                        end
+  # Find and compare the new number of In Progress licences:
+  table_content = @front_app.digitise_review_page.licence_table.text
+  status_count_after = table_content.scan(/(?=#{status})/).count
```
see above comment

andrewhick replied:
Done

andrewhick commented on features/step_definitions/digitise_steps.rb:
```diff
@@ -0,0 +1,131 @@
+Given(/^I reset a licence back to in progress$/) do
+
+  # Failsafe to prevent test running in production
+  @environment = Quke::Quke.config.custom["environment"].to_s
```
Good idea, I've added a helper to do this, which works.

andrewhick commented on features/digitise.feature:
```diff
@@ -0,0 +1,25 @@
+    And I reject the changes
+    Then the change is shown as "In progress"
+
+  Scenario: [WATER-1015] Review and approve licence changes
```
I can do this but it would involve repeating the background steps in the first two scenarios. Let me know what you think or if there's another way.

andrewhick commented:
Discussed with Tim and I'm going to merge as is, but think about how I could squash steps more in future so as to reduce the number of When Then When Then scenarios in the repository.

Cruikshanks committed:
Migrate abstraction alert tests

This change migrates the legacy water abstraction alert acceptance tests. These can be found in [cypress/integration/internal/waa](https://github.com/DEFRA/water-abstraction-ui/tree/main/cypress/integration/internal/waa).

Again as previously found, the existing tests contain a mix of testing the journey and _some_ validations. To make it easier to asses their value going forward the 2 have been split.

In this particular case the legacy code appeared to contain 2 distinct tests; `gauging-stations-and-water-abstraction-alerts-mbod.spec` and `gauging-stations-and-water-abstraction-alerts.spec`. However, after review they are almost identical. The only difference is in the choice of unit of measure for the threshold, **mBOD** or **Ml/d**. Whichever you select makes no difference to the subsequent journey. So, as part of the migration we have dropped the test for **Ml/d** as it is adding no value.

Cruikshanks committed:
Merge branch 'main' into migrate-abstraction-alerts
